### PR TITLE
Ensure that events are sorted correctly on the registrations page.

### LINKED ISSUE #1425: Events don't sort by event.rank in competitiors/registrations page
See the following screenshots.
image
image

### PR DESCRIPTION
**Pull request by jfly**
This fixes #1425.

### CODE REVIEW

File: WcaOnRails/app/controllers/registrations_controller.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@ class RegistrationsController < ApplicationController
 
   private def competition_from_params
     competition = if params[:competition_id]
-                    Competition.includes(:events).find(params[:competition_id])
+                    Competition.find(params[:competition_id])
```
Comment (jfly) [inline WcaOnRails/app/controllers/registrations_controller.rb]: More gotchas with `includes` =(. Check this out:

```ruby
irb(main):004:0> Competition.first.events.map(&:id)
  Competition Load (0.5ms)  SELECT  `Competitions`.* FROM `Competitions` ORDER BY `Competitions`.`id` ASC LIMIT 1
  Event Load (0.5ms)  SELECT `Events`.* FROM `Events` INNER JOIN `competition_events` ON `Events`.`id` = `competition_events`.`event_id` WHERE `competition_events`.`competition_id` = '1AVG2013' ORDER BY `Events`.`rank` ASC
=> ["333", "222", "444", "555", "666", "777", "333bf", "333oh", "minx", "pyram", "clock", "sq1"]
irb(main):005:0> Competition.includes(:events).first.events.map(&:id)
  Competition Load (0.6ms)  SELECT  `Competitions`.* FROM `Competitions` ORDER BY `Competitions`.`id` ASC LIMIT 1
  CompetitionEvent Load (0.5ms)  SELECT `competition_events`.* FROM `competition_events` WHERE `competition_events`.`competition_id` = '1AVG2013'
  Event Load (0.6ms)  SELECT `Events`.* FROM `Events` WHERE `Events`.`id` IN ('222', '333', '333bf', '333oh', '444', '555', '666', '777', 'clock', 'minx', 'pyram', 'sq1') ORDER BY `Events`.`rank` ASC
=> ["222", "333", "333bf", "333oh", "444", "555", "666", "777", "clock", "minx", "pyram", "sq1"]
```

Comment (jonatanklosko) [inline WcaOnRails/app/controllers/registrations_controller.rb]: That's funny, for me the second querying looks "way" more efficient!

File: WcaOnRails/spec/features/registration_spec.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+RSpec.feature "competition results" do
+  it "shows events in correct order" do
+    competition = FactoryGirl.create(:competition, :registration_open, :visible, events: Event.where(id: %w(222)))
+    # Add 333 after 222 in order to give 333 a higher id than 222, in an attempt to break event ordering.
+    competition.events = Event.where(id: %w(222 333))
+
+    visit competition_registrations_path(competition)
+
+    expect(page).to have_css('th.event-333 + th.event-222')
```
Comment (jonatanklosko) [inline WcaOnRails/spec/features/registration_spec.rb]: Not gonna work, see 6ab98167fa311da44b7bf66728b1272e8d24654c.

File: WcaOnRails/app/views/registrations/index.html.erb [hunk 3]
```diff
@@ -14,7 +14,7 @@
           <th class="country" data-sortable="true"><%= t 'common.user.citizen_of' %></th>
 
           <% @competition.events.each do |event| %>
-            <th class="<%= event.id %>">
+            <th class="event-<%= event.id %>">
```
Comment (larspetrus) [inline WcaOnRails/app/views/registrations/index.html.erb]: We just removed this class in PR #1455.

File: WcaOnRails/app/controllers/registrations_controller.rb (hunk 1 above)
Comment (larspetrus) [inline WcaOnRails/app/controllers/registrations_controller.rb]: The include is probably faster with a multi competition query.

Event constants in memory would be even faster :0

**Comment by jfly** [comment]
Updated in light of the changes in #1455.

**Comment by jfly** [comment]
I plan to merge this tomorrow (Tuesday) when I wake up if there are no objections.

**Pull request merged**